kcqrs-core/src/test/kotlin/com/clouway/kcqrs/core/SimpleAggregateRepositoryTest.kt

```diff
@@ -219,4 +291,103 @@ class SimpleAggregateRepositoryTest {
         }
     }
 
-}
\ No newline at end of file
+}
+
+class TestClassSnapshotMapper : SnapshotMapper<TestClass> {
+    private val gson = Gson()
+
+    override fun toSnapshot(data: TestClass): String {
```

Comment:
My proposal was this method signature to be changed to
```kotlin
toSnapshot(data: TestClass): Snapshot
```

where the Snapshot object will have to properties: payload and version. 


```kotlin
  // return version 1 of the snapshot
  return Snapshot(gson.toJson(mapped), 1)
```

Where this version is going to be persisted in the datastore and latter to be used for loading.

Reply:
.

Reply:
I could actually reuse the existing Snapshot. What do you think.

Reply:
I'm not sure that I understand what you mean.

Reply:
We have an existing class Snapshot that is accessible from here : 
```Kotlin
data class Snapshot(val version: Long, val data: Binary)
```

Reply:
Yes, it could be used.

Reply:
Implemented and removed the version incrementation from the EventStore. Now it will persist the one provided by the aggregate repository.